app/code/Magento/Downloadable/Test/Unit/Model/Sample/DeleteHandlerTest.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\Downloadable\Test\Unit\Model\Sample;
+
+use Magento\Catalog\Api\Data\ProductInterface;
+use Magento\Catalog\Model\Product;
+use Magento\Downloadable\Api\Data\SampleInterface;
+use Magento\Downloadable\Api\SampleRepositoryInterface as SampleRepository;
+use Magento\Downloadable\Model\Product\Type;
+use Magento\Downloadable\Model\Sample\DeleteHandler;
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager as ObjectManagerHelper;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Unit Test for \Magento\Downloadable\Model\Sample\DeleteHandler
+ */
+class DeleteHandlerTest extends TestCase
+{
+    const STUB_PRODUCT_TYPE = 'simple';
+    const STUB_PRODUCT_SKU = 'sku';
+    const STUB_SAMPLE_ID = 1;
+
+    /**
+     * @var ProductInterface|MockObject
+     */
+    private $entityMock;
+
+    /**
+     * @var SampleRepository|MockObject
+     */
+    private $sampleRepositoryMock;
+
+    /**
+     * @var DeleteHandler
+     */
+    private $deleteHandler;
+
+    /**
+     * @inheritDoc
+     */
+    protected function setUp()
+    {
+        $this->entityMock = $this->createMock(Product::class);
+        $this->sampleRepositoryMock = $this->getMockBuilder(SampleRepository::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['getList', 'delete'])
+            ->getMockForAbstractClass();
+
+        $this->deleteHandler = (new ObjectManagerHelper($this))->getObject(
+            DeleteHandler::class,
+            ['sampleRepository' => $this->sampleRepositoryMock]
+        );
+    }
+
+    /**
+     * Test case when provided Product has type Downloadable.
+     */
+    public function testExecuteWithDownloadableProduct()
```

Comment:
Another advice is to use return type hinting if we declare `strict types` in the future. So we have

```suggestion
    public function testExecuteWithDownloadableProduct(): void
```